src/components/Slider/Slider.tsx

```diff
@@ -103,41 +108,59 @@ export const Slider = React.forwardRef(function Slider(
                 marks={innerState.marks}
                 onBlur={onBlur}
                 onFocus={onFocus}
-                onChange={handleUpdate}
-                onChangeComplete={handleUpdateComplete}
+                onChange={setValue}
+                onChangeComplete={onUpdateComplete}
                 stateModifiers={stateModifiers}
-                // eslint-disable-next-line jsx-a11y/no-autofocus
                 autoFocus={autoFocus}
                 tabIndex={tabIndex}
-                data-qa={qa}
-                handleRender={
-                    hasTooltip
-                        ? (originHandle, handleProps) => {
-                              const styleProp = stateModifiers.rtl ? 'right' : 'left';
-                              return (
-                                  <React.Fragment>
-                                      {originHandle}
-                                      <SliderTooltip
-                                          value={handleProps.value}
-                                          className={b('tooltip')}
-                                          style={{
-                                              insetInlineStart:
-                                                  originHandle.props.style?.[styleProp],
-                                          }}
-                                          stateModifiers={stateModifiers}
-                                      />
-                                  </React.Fragment>
-                              );
-                          }
-                        : undefined
-                }
+                handleRender={(originHandle, handleProps) => {
+                    let handle: React.ReactElement = React.cloneElement(originHandle, {
+                        // @ts-expect-error originHandle has incorrect type, actually props is HTMLAttributes<HTMLDivElement>
+                        id: id ? `${id}-handle-${handleProps.index}` : undefined,
+                        'data-qa': qa ? `${qa}-handle-${handleProps.index}` : undefined,
+                    });
+                    if (name) {
+                        handle = (
+                            <React.Fragment>
+                                {handle}
+                                <input
+                                    ref={inputRef}
+                                    type="hidden"
+                                    name={name}
+                                    form={form}
+                                    value={handleProps.value}
+                                    disabled={disabled}
+                                />
```

Comment:
is it correct to put hidden input inside the handler? 
may be it will be better to move it outside the component and connect ti Slider thru Slider's props (onChange, value, etc)?

Reply:
> is it correct to put hidden input inside the handler? may be it will be better to move it outside the component and connect ti Slider thru Slider's state?

Also it is necessary to take into account the probability of two values appearing [number, number]. At my point it will be easier to manage such case.

For example like this:

```ts
...
const inputRef = useFormResetHandler({
    initialValue: Array.isArray(innerValue) ? innerValue[0] : innerValue,
    onReset: handleReset,
});
const inputSecondRef = useFormResetHandler({
    initialValue: Array.isArray(innerValue) ? innerValue[1] : innerValue,
    onReset: handleReset,
});
...
return (
        <div
            {...filterDOMProps(otherProps)}
            id={id}
            className={b(null, className)}
            ref={ref}
            style={style}
            data-qa={qa}
        >
            <input
                ref={inputRef}
                ... //
            />
            {Array.isArray(innerValue)? <input 
                ref={inputSecondRef}
                ... 
                /> : null}
            <div className={b('top', {size, hasTooltip})}></div>
            <BaseSlider
        ... />
        ...

```

Reply:
Normally, a slider component uses input with a type range under the hood, but not in this library. IMHO, integration with forms should be a base functionality for the slider component.

Reply:
@ValeraS , yes I agreed with you. And I suggest to put input near the BaseSlider instead of inside handle rendering function. I think it will be ore logical. 
Input still will be inside the Slider component. Because (in my opinion) input looks strange inside render prop.

Reply:
There is should be an input for each handle, so it's convenient to render the input alongside the handle.

Reply:
ok, thanks)